core/src/validator.rs

```diff
@@ -1925,85 +1929,106 @@ fn maybe_warp_slot(
     Ok(())
 }
 
+/// Searches the blockstore for data shreds with the incorrect shred version.
 fn blockstore_contains_bad_shred_version(
     blockstore: &Blockstore,
     start_slot: Slot,
-    shred_version: u16,
-) -> bool {
-    let now = Instant::now();
+    expected_shred_version: u16,
+) -> Result<bool, BlockstoreError> {
+    const TIMEOUT_MS: u64 = 60000;
+    let timer = AtomicInterval::default();
```

Comment:
why atomic? am i missing threads somewhere?

the compiler's going to insert memory fences around these ops which, while unlikely to be a problem at all here, unnecessarily limits optimization opportunities and we ought not be setting the precedent